lib/ukcpio/cpio.c

```diff
@@ -58,6 +58,44 @@ int uk_syscall_r_utime(const char *, const struct utimbuf *);
 int uk_syscall_r_mkdir(const char *, mode_t);
 int uk_syscall_r_symlink(const char *, const char *);
 int uk_syscall_r_stat(const char *, struct stat *);
+int uk_syscall_r_unlinkat(int, const char *, int);
+int uk_syscall_r_rename(const char *, const char *);
+
+static int
+try_rm_nonempty_dir(const char *path)
+{
+	/* rm -rf is nontrivial and costly, so we rename for speed */
+	/* TODO: revisit if efficient recursive dir removal is available */
+	int r;
+	char newpath[PATH_MAX];
+	char *newend = newpath + strlcpy(newpath, path, PATH_MAX);
+
+	if (unlikely(newend - newpath + 2 > PATH_MAX)) {
+		uk_pr_err("Cannot rename %s, path too long\n", path);
+		return -ENAMETOOLONG;
+	}
+	strcpy(newend, ".0");
+	r = uk_syscall_r_rename(path, newpath);
+	uk_pr_info("rename '%s' to '%s': %d\n", path, newpath, r);
```

Comment:
Capital letter for "Rename...". Unless you meant it as a "<syscall_name>..." format... in which case I would add a `()` to make it clear, i.e. `"rename() '%s'...` but I would prefer the former as it provides no additional value considering the dbg level messages already say the syscall name and its args